#166265846 Send welcome email to user on signup

OvieMudi opened the pull request:
#### What does this PR do?
Adds feature where email is sent to user on signup

#### Description of Task to be completed?
After successful signup, ensure an email sent to provided receiver email

#### How should this be manually tested?
To test manually on your local computer, _run the following commands_.

```
# Clone this repository
$ git clone https://github.com/emp-daisy/HelloBooksLib.git

# Navigate to the application folder
$ cd HelloBooksLib

# Checkout the branch
$ git checkout ft/166265846/send-email-to-user-after-signup

# Install dependencies
$ npm install

# Start the app in development mode
$ npm run dev
```

- make a GET request to  ` localhost:3000/api/v1/auth/signup`
- sign up with a valid email
- ensure an email is sent to the signup email

NB: see screenshot below on how the layout should look
#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#166265846](https://www.pivotaltracker.com/story/show/166265846)

#### Screenshots
![send_email1](https://user-images.githubusercontent.com/42499892/58749645-5e0b7400-8480-11e9-90fa-b27945d89095.PNG)

OvieMudi committed:
feat(email): add feature where email is sent to user on signup

- install and configure send-grid package
- add mailer helper class for sending messages
- add method for sending email on signup

[Delivers #166265846]

OvieMudi committed:
- update elint to include parsing options
- change sender email

chuxmykel commented on package.json:
```diff
@@ -5,10 +5,10 @@
     "start": "node build/index.js",
-    "pretest": "export NODE_ENV=test && npm run migrate:undo && npm run migrate",
-    "test": "export NODE_ENV=test && jest --forceExit --verbose",
+    "pretest": "cross-env NODE_ENV=test npm run migrate:undo && cross-env NODE_ENV=test npm run migrate",
```
Nice one. I see why this guy wasn't working now. Good job

chuxmykel commented on server/controllers/userController.js:
```diff
@@ -28,6 +29,8 @@ const UserController = {
             const token = auth.generateToken(user);
 
               models.User.create(user).then((createdUser) => {
```
I thought we agreed to use async/await??

OvieMudi replied:
async/await will require me to use try/catch just to access & log errors. I thought it's an overkill. Do you recommend I change it?

encodedBicoding replied:
I actually did mine this way you did your initially, i felt the same thing too.

cvjude replied:
I think we should have the same convention
Secondly you could wrap the whole function in an async await

encodedBicoding replied:
Okay Ovie, please adjust this, so we can merge

cvjude replied:
Ovie also has to do a rebase to clear all the conflicts on the base branch

encodedBicoding replied:
Jude i just noticed that there is a .vscode folder that vame with your PR, kindle ignore it

OvieMudi committed:
Feedback review changes and code refactor.

OvieMudi commented on server/helpers/mailer.js:
```diff
@@ -0,0 +1,78 @@
+   * @param {String} html - Email body
+   * @returns  {Object} - Mailer response
+   */
+  static sendMail(to, from, subject, html) {
```
We're supposed to convert this to async/await method.
Then we can convert the `.then` to `try/catch`

chuxmykel commented on server/controllers/userController.js:
```diff
@@ -26,30 +22,25 @@ const UserController = {
-            })
+            
+            try {
+                const USER = await models.Users.create(user);
```
USER or user?

chuxmykel commented on server/controllers/userController.js:
```diff
@@ -26,30 +22,25 @@ const UserController = {
+                }
+                return util.successStatus(res, 201, 'User added successfully', data)
+            } catch(err){
+                return util.errorStatus(res, 500, 'Intername error');
```
Internal.

OvieMudi committed:
Feedback changes on Mailer.js

I tried looking for how to use the async/await with send grid API, but didn't seem to find any. I hope my implementation works well.

OvieMudi committed:
Changes made according to feedback.

ocranbillions commented on server/controllers/userController.js:
```diff
@@ -1,21 +1,17 @@
-
             const hashPassword = auth.hashPassword(req.body.password);
 
             const user = {
```
@OvieMudi @encodedBicoding Guys, pls check lines 10 and 17. const user is declared twice within the same scope

encodedBicoding replied:
This has been taken care of.

encodedBicoding committed:
Feedback review change #4

encodedBicoding committed:
Feedback review change #4

encodedBicoding committed:
Fix merge conflict

encodedBicoding reviewed (commented):
Guys, please ignore the added file that had my work in it, I had to get the branch and work on it from my local machine, just so I could try to fix the merge conflict. that didn't work, I think Ovie has to be the one to fix that. Let me know if there are any minor changes to be made.

chuxmykel commented:
@encodedBicoding Has the merge conflict been fixed?
